src/flashbar/utils.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { FlashbarProps, FlashType } from './interfaces';
+
+// Since the position of a notification changes when the Flashbar is collapsed,
+// it is useful on some situations (e.g, for animating) to know the original position of the item
+// in the non-collapsed state.
+export interface StackableItem extends FlashbarProps.MessageDefinition {
+  expandedIndex: number;
+  collapsedIndex?: number;
+}
+
+const typesToColors: Record<FlashType, string> = {
+  error: 'red',
+  info: 'blue',
+  progress: 'blue',
+  success: 'green',
+  warning: 'blue',
+};
+
+function getColorFromType(type?: FlashType): string {
+  const defaultColor = 'blue';
+  return type ? typesToColors[type] || defaultColor : defaultColor;
+}
+
+export function getItemType(item: FlashbarProps.MessageDefinition) {
+  if (item.loading) {
+    return 'progress';
+  } else {
+    return item.type || 'info';
+  }
+}
+
+function getItemColor(item: FlashbarProps.MessageDefinition) {
+  return getColorFromType(getItemType(item));
+}
+
+/*
+ Returns a selection of notifications, preserving the order when possible but making sure that all different colors in
+ the stack are represented in the returned array.
+ The order corresponds to how they are represented when using the collapsible feature (newest first, oldest last).
+ */
+export function getItemSelection(
```

Comment:
I find the name of this function confusing. "Selecting" has a very specific meaning in our system and code base. I would suggest something like `groupItemsByColor`.

Reply:
OK, though not sure about the `group` part because it returns a single flat array and they are not grouped in a key-value object or anything like that. But I'll think of something else.

Reply:
Went for `getVisibleCollapsedItems`